openedx/core/djangoapps/content_libraries/models.py

```diff
@@ -127,3 +202,248 @@ def save(self, *args, **kwargs):  # lint-amnesty, pylint: disable=arguments-diff
     def __str__(self):
         who = self.user.username if self.user else self.group.name
         return f"ContentLibraryPermission ({self.access_level} for {who})"
+
+
+class LtiProfileManager(models.Manager):
+    """
+    Custom manager of LtiProfile mode.
+    """
+
+    def get_from_claims(self, *, iss, aud, sub):
+        """
+        Get the an instance from a LTI launch claims.
+        """
+        return self.get(platform_id=iss, client_id=aud, subject_id=sub)
+
+    def get_or_create_from_claims(self, *, iss, aud, sub):
+        """
+        Get or create an instance from a LTI launch claims.
+        """
+        try:
+            return self.get_from_claims(iss=iss, aud=aud, sub=sub)
+        except self.model.DoesNotExist:
+            # User will be created on ``save()``.
+            return self.create(platform_id=iss, client_id=aud, subject_id=sub)
+
+
+class LtiProfile(models.Model):
+    """
+    Content Libraries LTI's profile for Open edX users.
+
+    Unless Anonymous, this should be a unique representation of the LTI subject
+    (as per the client token ``sub`` identify claim) that initiated an LTI
+    launch through Content Libraries.
+    """
+
+    objects = LtiProfileManager()
+
+    user = models.OneToOneField(
+        get_user_model(),
+        null=True,
+        on_delete=models.CASCADE,
+        related_name='contentlibraries_lti_profile',
+        verbose_name=_('open edx user'),
+    )
+
+    platform_id = models.CharField(
```

Comment:
Can we rename `platform_id` to `issuer` for consistency? I found it a bit confusing at first to figure out what `platform_id` was until I saw that in almost every place it's referenced, it's used like `platform_id=iss`

Or since `platform_id` and `client_id` are both in [`LtiTool`](https://github.com/dmitry-viskov/pylti1.3/blob/ec931d3117889b7236048b3426fc9af0321b7bc6/pylti1p3/contrib/django/lti1p3_tool_config/models.py#L40-L43), would it be helpful to replace these two fields with a foreign key to `LtiTool`? (Or if the issuer/client_id are changed, do we want to ensure that old user records do not get associated with the new ID?)